### PR TITLE
Add unit tests for RenumberFanart rollback, indexed image push, and batch delete sync

### LINKED ISSUE #451: Add unit tests for RenumberFanart rollback, indexed image push, and batch delete sync
[mode: direct]
[model: sonnet]
[effort: medium]

## Description

PR #445 (issue #441) added fanart CRUD operations (RenumberFanart, batch delete with platform sync, indexed image push). Code review identified test gaps for three areas that need coverage to prevent regressions.

## Acceptance Criteria

- [ ] Unit tests for `RenumberFanart` in `internal/image/fanart_test.go`:
  - Phase 1 failure triggers rollback of already-staged files
  - Phase 2 failure triggers rollback of finalized files back to tmp, then restores originals
  - Happy path: survivors are correctly renumbered to contiguous indices
  - Edge cases: empty survivors slice, single survivor, mixed extensions
- [ ] Mock-based test for `UploadImageAtIndex` path in `handlePushImages` (`internal/api/handlers_push.go`):
  - Verify fanart images are uploaded at correct indices
  - Verify read failures are logged server-side and produce sanitized client errors
  - Verify upload failures produce sanitized client errors (no raw error leakage)
- [ ] Integration coverage for `syncAllFanartToPlatforms` in batch delete flow (`handleFanartBatchDelete`):
  - Verify sync is called after batch delete + renumber
  - Verify sync warnings are propagated to the response

## Additional Context

Identified during code review of PR #445. See review agents' findings: RenumberFanart criticality 9/10, UploadImageAtIndex 8/10, syncAllFanartToPlatforms 8/10.

Related: #441

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

- Add 8 new tests covering three high-criticality gaps identified during PR #445 review:
  - `RenumberFanart` Phase 1 and Phase 2 rollback (content restoration, temp file cleanup)
  - `handlePushImages` fanart indexed upload, read failure sanitization, upload failure sanitization
  - `handleFanartBatchDelete` platform sync after delete, sync warning propagation, renumber failure skips sync
- Fix pre-existing test safety issues in the same files:
  - Replace `t.Errorf` calls from httptest server goroutines with channel-capture pattern (race safety)
  - Add drain checks on sync channels to detect unexpected extra platform API calls
  - Strengthen HTTP/2 assertion from `t.Log` to `t.Error`

Closes #451
Deferred: #458 (injectable `FileRemover` for permission-based tests that skip as root)

## Test plan

- [x] `go test -count=1 ./internal/image/` -- RenumberFanart rollback tests pass
- [x] `go test -count=1 ./internal/api/` -- push and batch delete tests pass
- [x] `go test -count=1 -timeout 5m ./...` -- full suite green
- [x] `gofmt -d` -- no formatting issues
- [x] Pre-commit hooks pass (typos, gofmt, build, golangci-lint, govulncheck)
- [x] `/pr-review-toolkit:review-pr` -- all critical and important findings fixed

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/api/handlers_push_test.go [hunk 1]
```diff
@@ -30,12 +36,15 @@ func addTestConnectionWithURL(t *testing.T, r *Router, id, name, connType, url s
...
 	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.Method != http.MethodDelete {
-			t.Errorf("method = %s, want DELETE", r.Method)
-		}
-		if !strings.Contains(r.URL.Path, "/Images/Primary") {
-			t.Errorf("unexpected path: %s", r.URL.Path)
+		select {
+		case captureCh <- capture{method: r.Method, path: r.URL.Path}:
+		default:
 		}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_push_test.go]: Fixed in 036498c. Increased buffer to 3, replaced bare `default` receive with `time.After(5s)` timeout, and added a drain check to catch unexpected extra platform requests.

File: internal/api/handlers_push_test.go [hunk 2]
```diff
@@ -204,12 +234,15 @@ func TestHandleDeletePushImage_ArtistNotFound(t *testing.T) {
...
 	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		if r.Method != http.MethodDelete {
-			t.Errorf("method = %s, want DELETE", r.Method)
-		}
-		if !strings.Contains(r.URL.Path, "/Images/Logo") {
-			t.Errorf("unexpected path: %s", r.URL.Path)
+		select {
+		case captureCh <- capture{method: r.Method, path: r.URL.Path}:
+		default:
 		}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_push_test.go]: Fixed in 036498c. Same fix applied: buffer 1->3, `default`->`time.After(5s)`, plus drain check.

File: internal/api/handlers_push_test.go [hunk 3]
```diff
@@ -269,3 +332,203 @@ func TestHandleDeletePushImage_PlatformIDNotStoredAndNotProvided(t *testing.T) {
...
+	// The error message should say "read failed", not leak the raw OS error.
+	if len(resp.Errors) == 0 {
+		t.Fatal("expected at least one error for unreadable fanart")
+	}
+	if !strings.Contains(resp.Errors[0], "read failed") {
+		t.Errorf("error should say 'read failed', got %q", resp.Errors[0])
+	}
+	// Must not contain raw error details like "is a directory".
+	if strings.Contains(resp.Errors[0], "is a directory") {
+		t.Errorf("error leaks raw OS message: %q", resp.Errors[0])
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/api/handlers_push_test.go]: Fixed in 25a3ba9. Now checks multiple OS error substrings (`"is a directory"`, `"no such file"`, `"permission denied"`) and the on-disk symlink target name (`"does-not-exist"`).
